integration/store_gateway_limits_hit_test.go

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+//go:build requires_docker
+// +build requires_docker
+
+package integration
+
+import (
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/grafana/e2e"
+	e2edb "github.com/grafana/e2e/db"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/integration/e2emimir"
+	"github.com/grafana/mimir/pkg/storegateway"
+	"github.com/grafana/mimir/pkg/util/globalerror"
+)
+
+const (
+	blockRangePeriod = 5 * time.Second
+	querierTag       = "querier"
+	orgID            = "test"
+	series1          = "series_1"
+	series2          = "series_2"
+	series3          = "series_3"
+	series4          = "series_4"
```

Comment:
[nit] The series constants are not required be defined globally. Can be moved inside `Test_MaxSeriesAndChunksPerQueryLimitHit`.